crates/emath/src/range.rs

```diff
@@ -49,12 +49,48 @@ impl Rangef {
         self.max - self.min
     }
 
+    /// The length of the range, i.e. `max - min`.
+    #[inline]
+    pub fn center(self) -> f32 {
+        self.min + (self.span() / 2.0)
+    }
+
+    /// Returns the similarity between `self` and `other`.
+    ///
+    ///  * negative if `a` is left of `b`
+    /// * positive if `a` is right of `b`
+    /// * zero if the ranges overlap significantly
+    #[inline]
+    pub fn similarity(self, other: Rangef) -> f32 {
```

Comment:
This is a heuristic specific to the keyboard problem and doesn't belong in this file